packages/web3-shared/evm/src/libs/ConnectionContext.ts

```diff
@@ -108,9 +108,10 @@ export class ConnectionContext {
             ...this.payloadEditor.config,
             ...this._options?.overrides,
             from: this._options?.overrides?.from || this.payloadEditor.config?.from,
-            chainId: this._options?.overrides?.from
-                ? Number.parseInt(this._options?.overrides?.from, 10)
-                : this.payloadEditor.config?.chainId,
+            chainId:
+                typeof this._options?.overrides?.chainId === 'string'
+                    ? Number.parseInt(this._options?.overrides?.chainId, 16)
```

Comment:
Why a radix 16 number?

Reply:
```
parseInt('0x89', 16) // 137
parseInt('0x89', 10) // 0
```

Reply:
Why were the options given a hex chain id? We created it.

Reply:
ok